plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1195,6 +1204,28 @@ private static void updateProperty(UpdateProperties updateProperties, String ice
         }
     }
 
+    private static void updatePartitioning(Table icebergTable, Transaction transaction, List<String> partitionColumns)
+    {
+        UpdatePartitionSpec updatePartitionSpec = transaction.updateSpec();
+        Set<String> existingPartitionFields = icebergTable.spec().fields().stream()
+                .map(PartitionField::name)
+                .collect(toImmutableSet());
+        if (partitionColumns.isEmpty()) {
+            existingPartitionFields.forEach(updatePartitionSpec::removeField);
+        }
+        else {
+            existingPartitionFields.stream()
+                    .filter(partitionFieldName -> !partitionColumns.contains(partitionFieldName))
```

Comment:
do we need to lower case /upper case the column names?

Reply:
Updated this to do all the matching based on PartitionField so casing should match